src/options.ts

```diff
@@ -139,10 +157,10 @@ class SenderOptions {
 
   max_name_len?: number;
 
-  log?:
-    | ((level: "error" | "warn" | "info" | "debug", message: string) => void)
-    | null;
-  agent?: Agent;
+  log?: Logger;
+  agent?: Agent | http.Agent | https.Agent;
+
+  legacy_http?: boolean;
```

Comment:
This option doesn't seem to be documented. Also, I wouldn't call the default HTTP client "legacy", so something like `stdlib_http` or `default_http` may be a better name.